### PR TITLE
code to generate data and seed database

### PR DESCRIPTION
**Pull request by ktseto**

### CODE REVIEW

File: server/data.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  const s = dateTime.getSeconds();
+  return `"${y}-${mon}-${d} ${h}:${min}:${s}"`;
+};
+
+/*
+ *  Generate data for the "inventory" table
+ */
+
+const partyTypes = {
+  // all: _.range(1, 21),
```
Comment (TonyWu5) [inline server/data.js]: maybe a more descriptive variable name for this function could help me understand its logics.

File: server/data.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  small: _.range(1, 3),
+  large: _.range(19, 21),
+  scattered: [1, 10],
+};
+
+// 2019-12-01 00:00:00 to 2020-03-31 23:30:00 in 30-min increments
+// values in milliseconds from 1970-01-01 00:00:00
```
Comment (TonyWu5) [inline server/data.js]: nice descriptive annotations like this for bigger block of functions are very helpful for code review and for your future self when you need to revisit the code.

File: server/data.js (hunk 1 above)
Comment (TonyWu5) [reply server/data.js]: also, judging from the commented out property of all: _.range(1,21), is small supposed be range (1, 19)?

File: server/data.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  counter: 2,
+};
+
+const pointsTypes = {
+  standard: 'NULL',
+  extra: 1000,
+};
+
+// [id] restaurant_id, party, avail_at, special_seating, points, quantity
+const inventoryData = [];
```
Comment (TonyWu5) [inline server/data.js]: maybe naming it "availableTimeSlots" is a bit more descriptive?

File: server/data.js (hunk 2 above)
Comment (ktseto) [reply server/data.js]: noted

File: server/data.js (hunk 3 above)
Comment (ktseto) [reply server/data.js]: sounds good
```suggestion
const availableTimeslots = [];
```

File: server/data.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+Object.values(partyTypes).forEach((partyRange) => {
+  Object.values(dateTimeTypes).forEach((dateTimeRange) => {
+    Object.values(quantityTypes).forEach((quantityRange) => {
+      Object.values(seatingTypes).forEach((seating) => {
+        Object.values(pointsTypes).forEach((points) => {
+          restId += 1;
+          partyRange.forEach((party) => {
+            dateTimeRange.forEach((availAt) => {
+              quantityRange.forEach((quantity) => {
+                inventoryData.push([restId, party, format(availAt), seating, points, quantity]);
```
Comment (ktseto) [inline server/data.js]: updated based on review above
```suggestion
                availableTimeslots.push([restId, party, format(availAt), seating, points, quantity]);
```

File: server/data.js (hunk 3 above)
Comment (ktseto) [reply server/data.js]: sounds good
```suggestion
const availableTimeslots = [];
```

File: server/data.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+// 1 reservation already made for every 10th inventory type
+// booked_at can be way before when the inventory is available
+const reservationsData = [];
+
+_.range(1, inventoryData.length + 1, 10).forEach((invId) => {
+  const bookedAt = new Date(firstDateTime - (1 + Math.random()) * (30 * 24 * 60 * 60 * 1000));
+  reservationsData.push([123, invId, format(bookedAt)]);
+});
+
+exports.inventoryData = inventoryData;
```
Comment (ktseto) [inline server/data.js]: updated based on review above
```suggestion
exports.inventoryData = availableTimeslots;
```

File: server/data.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+});
+
+/*
+ *  Generate data for the "reservations" table
+ */
+
+// [id] user_id, inventory_id, booked_at
+// 1 reservation already made for every 10th inventory type
+// booked_at can be way before when the inventory is available
+const reservationsData = [];
```
[the remark this answers is not in the document]
Comment (ktseto) [reply server/data.js]: sounds good
```suggestion
const reservedTimeslots = [];
```

File: server/data.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+_.range(1, inventoryData.length + 1, 10).forEach((invId) => {
```
Comment (ktseto) [inline server/data.js]: updated based on review above
```suggestion
_.range(1, availableTimeslots.length + 1, 10).forEach((invId) => {
```

File: server/data.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  const bookedAt = new Date(firstDateTime - (1 + Math.random()) * (30 * 24 * 60 * 60 * 1000));
+  reservationsData.push([123, invId, format(bookedAt)]);
```
Comment (ktseto) [inline server/data.js]: updated based on review above
```suggestion
  reservedTimeslots.push([123, invId, format(bookedAt)]);
```

File: server/data.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+});
+
+exports.inventoryData = inventoryData;
+exports.reservationsData = reservationsData;
```
Comment (ktseto) [inline server/data.js]: updated based on review above
```suggestion
exports.reservationsData = reservedTimeslots;
```

File: server/data.js (hunk 1 above)
Comment (ktseto) [reply server/data.js]: unclear how to improve clarity of the object name... it describes small parties, large parties, etc.  also, small parties are defined as parties of 1-3.  Parties of 19 would be large.  leave the names as-is.

**Pull request merged**